include/openmc/settings.h

```diff
@@ -56,7 +58,11 @@ extern bool ufs_on;                   //!< uniform fission site method on?
 extern bool urr_ptables_on;           //!< use unresolved resonance prob. tables?
 extern bool write_all_tracks;         //!< write track files for every particle?
 extern bool write_initial_source;     //!< write out initial source file?
+extern bool weightwindow_on;          //!< use weight window or not
 
+// weight window   
+extern std::shared_ptr<weight_window::WeightWindow>  ww_settings;  //!< weight window settings
```

Comment:
This makes more sense as a `unique_ptr`. No other `shared_ptr` is getting created that refers to this, especially since this is at global scope. Moreover, you can remove the `weightwindow_on` setting. Rather, we can just check if `ww_settings` is a nullptr and assume weight windows are not turned on if so.